### ISSUE TITLE
[FEATURE] -  Support pagination for PPL and SQL query

### ISSUE THREAD

**Issue by penghuo** [opensearch-project/sql#656]
**Is your feature request related to a problem?**
* [The new engine fetches a default size of index from OpenSearch set by this setting, the default value is 200](https://github.com/opensearch-project/sql/blob/main/docs/user/admin/settings.rst#plugins-query-size-limit). It is a blocker for ML plugin to pull more data by using PPL.
* [Cursor only support in V1 engine. ](https://github.com/opensearch-project/sql/blob/main/docs/user/interfaces/endpoint.rst#cursor). We should support it in V2 engine.

**What solution would you like?**

- [ ] add pagination support for operator which will unblock ML plugin use case.
- [ ] add cursor support for V2 engine.

**Comment by vamsi-amazon** [comment]
In the second cursor request can there be a scenario with some extra calculations along with push down.  how do you retain entire physical plan?

**Comment by seankao-az** [comment]
> In the second cursor request can there be a scenario with some extra calculations along with push down. how do you retain entire physical plan?

Thanks for pointing this out. I've come up with a design idea that takes this into consideration. It also avoids many of the problems faced in the PoC PR #693 
* The design in #693 uses the the `scrollId` returned from `OpenSearchScrollRequest` as the `cursor` for the sql/ppl request. New design hides it and exposes a new cursor we generated, so we no longer need to break `PhysicalPlan` interface to get the `scrollId`.
* retains the physical plan for extra calculations

**Comment by penghuo** [comment]
relate to https://github.com/opensearch-project/sql/issues/947

**Comment by rrlamichhane** [comment]
What's the status of this?

**Comment by Yury-Fridlyand** [comment]
WIP, you can track it in https://github.com/Bit-Quill/opensearch-project-sql/pull/226

**Comment by Yury-Fridlyand** [comment]
https://user-images.githubusercontent.com/88679692/224208630-8d38d833-abf8-4035-8d15-d5fb4382deca.mp4

**Comment by Yury-Fridlyand** [comment]
https://github.com/opensearch-project/sql/assets/88679692/f21754d0-c228-4959-b38a-736b36c4a2c2

**Comment by Yury-Fridlyand** [comment]
- [x] - Pagination basement - support `select * from <table>` #1666
- [x] - Support `WHERE` clause - #1500 
- [x] - Support `ORDER BY` clause - #1599
- [ ] - Support `LIMIT` clause - WIP
- [ ] - Support system queries - tracked by #1712 
- [ ] - Support `OFFEST` clause
- [ ] - Support aggregation and `GROUP BY` clause
- [ ] - Support `NESTED` function - tracked by #1718
- [ ] - Support pagination in PPL

**Comment by Yury-Fridlyand** [comment]
Superseded by #1759
